zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -489,4 +493,116 @@ private ExporterResourceProvider mockResourceProvider(
 
     return provider;
   }
+
+  @Nested
+  class ImportersCompletedTests {
+    private final ExporterTestController controller = spy(new ExporterTestController());
+    private final CamundaExporter camundaExporter = new CamundaExporter();
+    private final int partitionId = 1;
+    private final String importPositionIndexName =
+        new ImportPositionIndex(CONFIG_PREFIX, true).getFullQualifiedName();
+
+    @BeforeEach
+    void setup() {
+      controller.resetScheduledTasks();
+    }
+
+    @TestTemplate
+    void shouldNotFlushIfImportersAreNotCompleted(
+        final ExporterConfiguration config, final SearchClientAdapter clientAdapter)
+        throws IOException {
+      // given
+      final var context = spy(getContextFromConfig(config));
+      doReturn(partitionId).when(context).getPartitionId();
+      camundaExporter.configure(context);
+      camundaExporter.open(controller);
+
+      // when
+
+      // adds a not complete position index document so exporter sees importing as not yet completed
+      indexImportPositionEntity(ImportValueType.DECISION, false, clientAdapter);
+      clientAdapter.refresh();
+
+      controller.runScheduledTasks(Duration.ofMinutes(1));
+
+      final var record =
+          factory.generateRecord(
+              ValueType.AUTHORIZATION,
+              r -> r.withBrokerVersion("8.7.0").withTimestamp(System.currentTimeMillis()));
+
+      camundaExporter.export(record);
+
+      // then
+      assertThat(controller.getPosition()).isEqualTo(-1);
+      verify(controller, never()).updateLastExportedRecordPosition(eq(record.getPosition()), any());
+
+      final var authHandler =
+          getHandlers(config).stream()
+              .filter(handler -> handler.getHandledValueType().equals(record.getValueType()))
+              .filter(handler -> handler.handlesRecord(record))
+              .findFirst()
+              .orElseThrow();
+      final var recordId = authHandler.generateIds(record).getFirst();
+
+      assertThat(
+              clientAdapter.get(recordId, authHandler.getIndexName(), authHandler.getEntityType()))
+          .isNull();
+    }
+
+    @TestTemplate
+    void shouldFlushIfImportersAreCompleted(
+        final ExporterConfiguration config, final SearchClientAdapter clientAdapter)
```

Comment:
I added the test the failing when we reach the bulk size, not sure the second test is needed as exporting is idempotent, also I'm not sure why it would write twice? Each `writer.addRecord` call will only add one set of cached entities no matter how many times you call it for a record, and if a record has `export` called on it and succeeds the retry logic shouldn't kick in.